src/ml-hooks.tsx

```diff
@@ -20,24 +20,25 @@ export const useMlActions = () => {
   return actions;
 };
 
+export interface PredictionResult {
+  confidences: Confidences;
+  detected: Gesture | undefined;
+}
+
 export const usePrediction = () => {
   const buffer = useBufferedData();
   const logging = useLogging();
   const [status] = useMlStatus();
   const [connectStatus] = useConnectStatus();
   const connection = useConnectActions();
-  const [confidences, setConfidences] = useState<Confidences | undefined>();
+  const [prediction, setPrediction] = useState<PredictionResult | undefined>();
   const [gestureData] = useGestureData();
 
-  // Avoid re-renders due to threshold changes which update gestureData.
-  // We could consider storing them elsewhere, perhaps with the model.
-  const classificationIdsRecalculated = gestureData.data.map((d) => d.ID);
-  const classificationIdsKey = JSON.stringify(classificationIdsRecalculated);
-  const classificationIds: number[] = useMemo(
-    () => JSON.parse(classificationIdsKey) as number[],
-    [classificationIdsKey]
-  );
-
+  // Use a ref to prevent restarting the effect every time thesholds change.
+  // We only use the ref's value during the setInterval callback not render.
+  // We can avoid this by storing the thresolds separately in state, even if we unify them when serializing them.
+  const gestureDataRef = useRef(gestureData);
+  gestureDataRef.current = gestureData;
```

Comment:
I've swapped one form of evil for another. I think it's OK as it doesn't affect the returned JSX but still want to restructure the storage so this is "model" stage rather than "data" state.